tree_assume/src/optimizations/is_valid.rs

```diff
@@ -40,11 +40,12 @@ fn test_is_valid() -> Result<(), egglog::Error> {
     let myloop = dowhile(
         assume(inlet(int(2)), single(int(1))),
         parallel!(
-            less_than(get(arg(intt()), 0), int(3)),
+            less_than(get(arg(), 0), int(3)),
             get(switch!(int(0); parallel!(int(4), int(5))), 0)
         ),
-    );
-    let not_made_valid = sub(arg(intt()), arg(intt())); // this expression is indeed valid, but won't be marked as so
+    )
+    .with_arg_types(emptyt(), tuplet!(intt()));
+    let not_made_valid = sub(arg(), arg()).with_arg_types(intt(), intt()); // this expression is indeed valid, but won't be marked as so
```

Comment:
what does this comment mean? (ik not added by this diff)

Reply:
clarified-
```
// this expression is valid (it uses only IR constructors)
    // but it isn't a sub-expression of the initial one, so it won't be
    // marked as valid.```